src/ValidationExemption.php

```diff
@@ -0,0 +1,247 @@
+	 *
+	 * The difference here with `data-amp-unvalidated-tag` is that the PX-verified means that the tag will work
+	 * properly Bento components and CSS tree shaking.
+	 *
+	 * @var string
+	 */
+	const PX_VERIFIED_TAG_ATTRIBUTE = 'data-px-verified-tag';
+
+	/**
+	 * HTML attribute to indicate one or more attributes are exempted from AMP validation.
+	 *
+	 * @var string
+	 */
+	const AMP_UNVALIDATED_ATTRS_ATTRIBUTE = 'data-amp-unvalidated-attrs';
+
+	/**
+	 * HTML attribute to indicate an tag/element is exempted from AMP validation.
+	 *
+	 * @var string
+	 */
+	const AMP_UNVALIDATED_TAG_ATTRIBUTE = 'data-amp-unvalidated-tag';
+
+	/**
+	 * Check whether PX is verified for node.
+	 *
+	 * This means that it is exempted from AMP validation, but it doesn't mean the PX is negatively impacted.
+	 *
+	 * @param DOMNode $node Node.
+	 * @return bool Whether node marked as being PX-verified.
+	 */
+	public static function is_px_verified_for_node( DOMNode $node ) {
+
+		// Check element.
+		if (
+			$node instanceof Element
+			&&
+			$node->hasAttribute( self::PX_VERIFIED_TAG_ATTRIBUTE )
+		) {
+			return true;
+		}
+
+		// Check attribute.
+		if (
+			$node instanceof DOMAttr
+			&&
+			self::check_for_attribute_token_list_membership( $node, self::PX_VERIFIED_ATTRS_ATTRIBUTE )
+		) {
+			return true;
+		}
+
+		return false;
+	}
+
+	/**
+	 * Mark node as being PX-verified.
+	 *
+	 * @param DOMNode $node Node.
+	 * @return bool Whether successful.
+	 */
+	public static function mark_node_as_px_verified( DOMNode $node ) {
+
+		if ( $node instanceof Element ) {
+			if ( ! $node->hasAttribute( self::PX_VERIFIED_TAG_ATTRIBUTE ) ) {
+				if ( ! $node->ownerDocument ) {
+					return false; // @codeCoverageIgnore
+				}
+				$node->setAttributeNode( $node->ownerDocument->createAttribute( self::PX_VERIFIED_TAG_ATTRIBUTE ) );
+			}
+			return true;
+		}
+
+		if ( $node instanceof DOMAttr ) {
+			$element = $node->parentNode;
+			if ( ! $element instanceof Element ) {
+				return false; // @codeCoverageIgnore
+			}
+
+			$attr_value = $element->getAttribute( self::PX_VERIFIED_ATTRS_ATTRIBUTE );
+			if ( $attr_value ) {
+				$attr_value .= ' ' . $node->nodeName;
+			} else {
+				$attr_value = $node->nodeName;
+			}
+
+			$element->setAttribute( self::PX_VERIFIED_ATTRS_ATTRIBUTE, $attr_value );
+			return true;
+		}
+
+		return false;
+	}
+
+	/**
+	 * Mark node as not being AMP-validated.
+	 *
+	 * @param DOMNode $node Node.
+	 * @return bool Whether successful.
+	 */
+	public static function mark_node_as_amp_unvalidated( DOMNode $node ) {
+
+		if ( $node instanceof Element ) {
+			if ( ! $node->hasAttribute( self::AMP_UNVALIDATED_TAG_ATTRIBUTE ) ) {
+				if ( ! $node->ownerDocument ) {
+					return false; // @codeCoverageIgnore
+				}
+				$node->setAttributeNode( $node->ownerDocument->createAttribute( self::AMP_UNVALIDATED_TAG_ATTRIBUTE ) );
+			}
+			return true;
+		}
+
+		if ( $node instanceof DOMAttr ) {
```

Comment:
If the `$node` was already detected to be an instance of `Element`, there's no point in checking it for `DOMAttr`.

```suggestion
		} elseif ( $node instanceof DOMAttr ) {
```